cpp/include/kvikio/defaults.hpp

```diff
@@ -57,7 +57,9 @@ inline bool getenv_or(std::string_view env_var_name, bool default_val)
   }
   // Convert to lowercase
   std::string str{env_val};
-  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+  // To avoid UB in std::tolower(), the character must be cast to unsigned char
```

Comment:
😮

Reply:
I've expanded the initial comment, since there's actually another issue on top of UB: `std::tolower` is not an addressable function, and passing it as a function pointer to the STL algorithm functions would cause the program behavior to be "[unspecified (possibly ill-formed)](https://eel.is/c++draft/namespace.constraints#namespace.std-6)". C++ is wild.

Reply:
> `std::tolower` is not an addressable function, and passing it as a function pointer to the STL algorithm functions would cause the program behavior to be "[unspecified (possibly ill-formed)](https://eel.is/c++draft/namespace.constraints#namespace.std-6)". C++ is wild.

What in the (hello) world...!